lib/runtime/src/pipeline/network/egress/addressed_router.rs

```diff
@@ -154,33 +211,77 @@ where
 
         // TRANSPORT ABSTRACT REQUIRED - END HERE
 
-        tracing::trace!(request_id, "enqueueing two-part message to nats");
+        // Send request based on transport mode
+        match &self.req_transport {
+            RequestTransport::Nats(client) => {
+                tracing::trace!(request_id, "enqueueing two-part message to NATS");
 
-        // Insert Trace Context into Headers
-        // Enables span to be created in push_endpoint before
-        // payload is parsed
+                // Prepare trace headers using the OpenTelemetry injector pattern
+                // This handles traceparent and tracestate headers according to W3C Trace Context standard
+                let mut headers = HeaderMap::new();
+                inject_otel_context_into_nats_headers(&mut headers, None);
```

Comment:
We should follow the same pattern of injecting into headers - potentially we can create a common function to add into headers regardless of transport